clarity/contracts/arkadiko-auction-engine-v4-1.clar

```diff
@@ -0,0 +1,295 @@
+  (liquidation-rewards <liquidation-rewards-trait>)
+)
+  (let (
+    (vault (contract-call? .arkadiko-vault-data-v1-1 get-vault-by-id vault-id))
+  )
+    (asserts! (not (shutdown-activated)) (err ERR-EMERGENCY-SHUTDOWN-ACTIVATED))
+    (asserts! (is-eq (contract-of vault-manager) (unwrap-panic (contract-call? .arkadiko-dao get-qualified-name-by-name "freddie"))) (err ERR-NOT-AUTHORIZED))
+    (asserts! (is-eq (contract-of coll-type) (unwrap-panic (contract-call? .arkadiko-dao get-qualified-name-by-name "collateral-types"))) (err ERR-NOT-AUTHORIZED))
+    (asserts! (is-eq (contract-of oracle) (unwrap-panic (contract-call? .arkadiko-dao get-qualified-name-by-name "oracle"))) (err ERR-NOT-AUTHORIZED))
+    (asserts! (is-eq (contract-of reserve) (unwrap-panic (contract-call? .arkadiko-dao get-qualified-name-by-name "sip10-reserve"))) (err ERR-NOT-AUTHORIZED))
+    (asserts! (is-eq (get is-liquidated vault) false) (err ERR-AUCTION-NOT-ALLOWED))
+
+    (let (
+      (collateral-type (unwrap-panic (contract-call? vault-manager get-collateral-type-for-vault vault-id)))
+      (collateral-to-debt-ratio (unwrap-panic (contract-call? vault-manager calculate-current-collateral-to-debt-ratio vault-id coll-type oracle true)))
+      (liquidation-ratio (unwrap-panic (contract-call? coll-type get-liquidation-ratio collateral-type)))
+      (amounts (unwrap-panic (as-contract (contract-call? vault-manager liquidate vault-id coll-type))))
+
+      (auction-id (+ (var-get last-auction-id) u1))
+      (auction {
+        id: auction-id,
+        auction-type: "collateral",
+        vault-id: vault-id,
+        collateral-amount: (get ustx-amount amounts),
+        collateral-token: (get collateral-token vault),
+        collateral-address: (unwrap-panic (contract-call? coll-type get-token-address (get collateral-type vault))),
+        debt-to-raise: (get vault-debt amounts),
+        discount: (unwrap-panic (contract-call? coll-type get-liquidation-penalty (get collateral-type vault))),
+        total-debt-burned: u0,
+        total-collateral-sold: u0,
+        ended-at: u0,
+      })
+    )
+      (asserts! (>= liquidation-ratio collateral-to-debt-ratio) (err ERR-NO-LIQUIDATION-REQUIRED))
+
+      ;; Add auction
+      (map-set auctions { id: auction-id } auction)
+
+      ;; Try to burn
+      (try! (burn-usda auction-id oracle coll-type vault-manager ft reserve liquidation-pool liquidation-rewards))
+
+      (print { type: "auction", action: "created", data: auction })
+      (ok true)
+    )
+  )
+)
+
+;; @desc burn USDA for vault and get part of collateral
+;; @param auction-id; the auction
+;; @param oracle; oracle
+;; @param coll-type; collateral types
+;; @param vault-manager; vault manager
+;; @param ft; collateral token if not STX vault
+;; @param reserve; reserve to get collateral tokens from
+;; @param liquidation-pool; pool to get USDA from
+;; @param liquidation-rewards; contract to deposit rewards
+(define-public (burn-usda
```

Comment:
Q: should this be publicly callable?

Reply:
This is useful when there was not enough USDA in the pool to fully liquidate the vault when starting the auction. So part of the liquidation can be done afterwards.